docs/PARSING_DATA.md

```diff
@@ -0,0 +1,216 @@
+there are three different parsers capable of reading JSON data:
+
+* `json`: This parser is great for flat JSON data. If your JSON is more complex
+  and for example, has other objects or nested arrays, then do not use this and
+  look at the other two options.
+* `json_v2`: The v2 parser was created out of the need to parse JSON objects. It
+  can take on more advanced cases, at the cost of additional configuration.
+* `xpath`: The xpath parser is the most capable of the three options. While the
+  xpath name may imply XML data, it can parse a variety of data types using
+  XPath expressions.
+
+## Tags and fields
+
+The next step is to look at the data and determine how the data needs to be
+split up between tags and fields. Tags are generally strings or values that a
+user will want to search on. While fields are the raw data values, numeric
+types, etc. Generally, data is considered to be a field unless otherwise
+specified as a tag.
+
+## Timestamp
+
+To parser a timestamp, at the very least the users needs to specify which field
+has the timestamp and what the format of the timestamp is. The format can either
+by a predefined Unix timestamp or parsed using a custom format based on Go
+reference time.
+
+For Unix timestamps Telegraf understand the following settings:
+
+| Timestamp             | Timestamp Format |
+|-----------------------|------------------|
+| `1709572232`          | `unix`    |
+| `1709572232123`       | `unix_ms` |
+| `1709572232123456`    | `unix_us` |
+| `1709572232123456789` | `unix_ns` |
+
+To use a custom timestamp, a user must provide the [Go reference time][]
+notation. Here are a few example timestamps and their Go reference time
+equivalent:
+
+| Timestamp                     | Go reference time             |
+|-------------------------------|-------------------------------|
+| `2024-03-04T17:10:32`         | `2006-01-02T15:04:05` |
+| `04 Mar 24 10:10 -0700`       | `02 Jan 06 15:04 -0700` |
+| `2024-03-04T10:10:32Z07:00`   | `2006-01-02T15:04:05Z07:00` |
+| `2024-03-04 17:10:32.123+00`  | `2006-01-02 15:04:05.999+00` |
+| `2024-03-04T10:10:32.123456Z` | `2006-01-02T15:04:05.000000Z` |
+| `2024-03-04T10:10:32.123456Z` | `2006-01-02T15:04:05.999999999Z` |
+
+Note for fractional second values, the user can use either a `9` or `0`. Using a
+`0` forces a certain length, but using `9`s do not.
+
+[Go reference time]: https://pkg.go.dev/time#pkg-constants
+
+## Examples
+
+Below are a few basic examples to get users started.
+
+### CSV
+
+Given the following data:
+
+```csv
+node,temp,humidity,alarm,time
+node1,32.3,23,false,2023-03-06T16:52:23Z
+node2,22.6,44,false,2023-03-06T16:52:23Z
+node3,17.9,56,true,2023-03-06T16:52:23Z
+```
+
+Here is corresponding parser configuration and result:
+
+```toml
+[[inputs.file]]
+files = ["test.csv"]
+data_format = "csv"
+
+csv_header_row_count = 1
+csv_column_names = ["node","temp","humidity","alarm","time"]
```

Comment:
Couldn't you leave this out in this case?

Reply:
I think so. I tried to consider what config options are most common and include those. Versus either listing all of them or only showing the bare minimum.

Reply:
I'm fine with keeping it as-is.